internal/epoch/epoch_manager.go

```diff
@@ -186,6 +186,8 @@ type Manager struct {
 	log      logging.Logger
 	timeFunc func() time.Time
 
+	allowCleanupWritesOnIndexLoad bool
+
```

Comment:
Is it ever possible to have `compact` property be `nil` while `allowCleanupWritesOnIndexLoad` is true ?

If not, then how about we use the presence of `compact` function property as the check within `allowWrites()` ? We should not need a dedicated `allowCleanupWritesOnIndexLoad` if it is exactly the same as `bool(compact)`. I think it is... given the below truth table holds:

```
+----------------------------------+-------------------+---------+
+ allowCleanupWritesOnIndexLoad    + compact function  +         +
+----------------------------------+-------------------+---------+
+ true                             + non-nil           +  ✓      +
+ true                             + nil               +  ⨉      +
+ false                            + non-nil           +  ⨉      +
+ false                            + nil               +  ✓      +
+----------------------------------+-------------------+---------+
```

`NewManager()` can then either be passed a nil `compact` or non-nil one based upon whether we want to allow it or not, keeping the `Manager` interface simpler. If we don't need to do compaction then don't pass a compaction function.

WDYT ?

Reply:
This was simplified in the upstream PRs. See kopia/kopia#3645